### PR TITLE
Adding animated page transitions

### PR DESCRIPTION
**Pull request by devnook**
+ Added back entry points for pages javascript
+ Moved existing js code to modules
+ Added router to load partials on navigation instead of reloading the whole page 

@rowan-m @samdutton  Let's discuss if this approach to partial navigation and modules loading is something we still want to do.

### CODE REVIEW

**Comment by samdutton** [comment]
Hi @devnook — could you fix the stuff that's causing the linting errors?

**Comment by samdutton** [comment]
I think the page transition animations are fine. 

My only reservation is that none of the big e-commerce sites are doing this. 

Also, I wonder if the animation actually takes away from the snappiness and makes the site feel slower: 
• [animation](https://drive.google.com/open?id=1yaEOyD4maYLQ9qGzX3H2okBgv4P8iK-4)  
• [no animation](https://drive.google.com/open?id=1N2ecoFBhcQT7Enm1NBZAa9903ryrrb07)

**Comment by samdutton** [comment]
Agree that we should discuss the approach to partial navigation and module loading.

Other changes LGTM.

File: src/client/js/router.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  async _onLinkClick(event) {
+    let target = event.target;
+
+    if (target.tagName !== 'A') {
+      target = this._getParentByTagName(target, 'A');
+      if (!target) return;
+    }
+    const link = new URL(target.href);
```
Comment (samdutton) [inline src/client/js/router.js]: I get this error when I click on the menu icon: 

`Uncaught (in promise) TypeError: Failed to construct 'URL': Invalid URL
    at Router._onLinkClick (router.js:63)`

Comment (devnook) [reply src/client/js/router.js]: Added additional check

**Comment by devnook** [comment]
PTAL

**Pull request merged**